src/backend/model_deployments/cohere_platform.py

```diff
@@ -90,11 +95,11 @@ async def invoke_chat_stream(
             )
             yield to_dict(event)
 
+    @collect_metrics_rerank
     async def invoke_rerank(
         self, query: str, documents: List[Dict[str, Any]], **kwargs: Any
     ) -> Any:
-        response = self.client.rerank(
-            query=query, documents=documents, model=DEFAULT_RERANK_MODEL
-        )
-
+        model = DEFAULT_RERANK_MODEL
+        add_rerank_model_to_request_state(model, **kwargs)
```

Comment:
I'm not sure if this is the best place to add the rerank model to the state since we would need to add the same snippet to all deployments. Wdyt about defaulting the model in the metrics service? If we ever decide to support model selection, we can follow up with this change.

Reply:
addressed